src/Cupboard.Tests/Unit/Providers/ChocolateyProviderTests.cs

```diff
@@ -63,8 +74,96 @@ public void Should_Not_Install_Package_If_Present()
+                var result = fixture.Run("-y");
+
+                // Then
+                result.Report.GetState<ChocolateyPackage>("Visual Studio Code").ShouldBe(ResourceState.Changed);
+                fixture.Logger.WasLogged("Installing Chocolatey package [yellow]vscode[/]");
+                fixture.Logger.WasLogged("The Chocolatey package [yellow]vscode[/] was installed");
+            }
+
+            [WindowsFact]
+            public void Should_Install_Package_If_Lower_Version()
+            {
+                // Given
+                var fixture = new CupboardFixture();
+                fixture.Security.IsAdmin = true;
+                fixture.Process.RegisterDefaultResult(new ProcessRunnerResult(0));
+                fixture.Configure(ctx => ctx.UseManifest<Manifests.InstallVersion>());
+
+                fixture.Process.Register("choco", "list --limit-output --local-only --exact vscode",
+                    new ProcessRunnerResult(0, "vscode|1.57.0\n"),
+                    new ProcessRunnerResult(0, "vscode|1.58.0\n"));
+
+                fixture.Process.Register("choco", "install vscode -y --version 1.58.0",
+                    new ProcessRunnerResult(0, "Lol installed VSCode"));
+
+                // When
+                var result = fixture.Run("-y");
+
+                // Then
+                result.Report.GetState<ChocolateyPackage>("Visual Studio Code").ShouldBe(ResourceState.Changed);
+                fixture.Logger.WasLogged("Installing Chocolatey package [yellow]vscode[/]");
+                fixture.Logger.WasLogged("The Chocolatey package [yellow]vscode[/] was installed");
+            }
+
+            [WindowsFact]
+            public void Should_Not_Install_Package_Of_Same_Version()
+            {
+                // Given
+                var fixture = new CupboardFixture();
+                fixture.Security.IsAdmin = true;
+                fixture.Configure(ctx => ctx.UseManifest<Manifests.InstallVersion>());
+
+                fixture.Process.Register("choco", "list --limit-output --local-only --exact vscode",
+                    new ProcessRunnerResult(0, "vscode|1.58.0\n"));
+
+                // When
+                var result = fixture.Run("-y");
+
+                // Then
+                result.Report.GetState<ChocolateyPackage>("Visual Studio Code").ShouldBe(ResourceState.Unchanged);
+                fixture.Logger.WasLogged("The Chocolatey package [yellow]vscode[/] is already installed");
+            }
+
+            [WindowsFact]
+            public void Should_Not_Install_Package_Of_Higher_Version()
+            {
+                // Given
+                var fixture = new CupboardFixture();
+                fixture.Security.IsAdmin = true;
+                fixture.Configure(ctx => ctx.UseManifest<Manifests.InstallVersion>());
+
+                fixture.Process.Register("choco", "list --limit-output --local-only --exact vscode",
+                    new ProcessRunnerResult(0, "vscode|1.59.0\n"));
```

Comment:
We could add option for `--allow-downgrade` to support downgrading versions.

Reply:
Sounds like a good idea!